analysis_runner/cli_analysisrunner.py

```diff
@@ -277,3 +251,99 @@ def _perform_shebang_check(script: str) -> None:
                 f', consider inserting "{suggestion_shebang}" at the top of this file'
             )
         raise AssertionError(message)
+
+
+def get_repository_specific_information(  # noqa: C901
+    repository: str | None,
+    commit: str | None,
+    branch: str | None,
+    cwd: str | None,
+    script: list[str],
+) -> RepositorySpecificInformation:
+    """Do all the repository specific stuff (like fetching repo, commit, cwd)
+
+    Args:
+        repository (str | None): _description_
+        commit (str | None): _description_
```

Comment:
I'd probably question the value of a doc comment here without extra descriptions, the types convey the same info.